### PR TITLE
Improve bls::SecretKey privacy

### PR DESCRIPTION
**Pull request by paulhauner**
## Issue Addressed

NA

## Proposed Changes

Ensures that `bls::SecretKey` only outputs bytes in a `Zeroize` wrapper and has the absolute minimum impls and public functions.

The security properties of `milagro_bls` are out of scope of this PR (I know they need some work).

### CODE REVIEW

**Comment by paulhauner** [comment]
@kirk-baird I've requested your review, if you don't mind and have the time? It's pretty simple, pretty much just removing code :)

File: crypto/bls/src/secret_key.rs [hunk 1]
```diff
@@ -42,52 +43,21 @@ impl SecretKey {
...
             3, 211, 210, 129, 231, 69, 162, 234, 16, 15, 244, 214, 126, 201, 0, 85, 28, 239, 82,
             121, 208, 190, 223, 6, 169, 202, 86, 236, 197, 218, 3, 69,
         ];
-        let original = SecretKey::from_bytes(&byte_key).unwrap();
-
-        let bytes = ssz_encode(&original);
-        let decoded = SecretKey::from_ssz_bytes(&bytes).unwrap();
-
-        assert_eq!(original, decoded);
+        SecretKey::from_bytes(&byte_key).unwrap();
```
Comment (kirk-baird) [inline crypto/bls/src/secret_key.rs:61]: Should we test the round trip?
```rust
        let key = SecretKey::from_bytes(&byte_key).unwrap();
        assert_eq!(byte_key, key.as_bytes().as_bytes());
```

File: validator_client/src/validator_store.rs [hunk 2]
```diff
@@ -15,12 +15,26 @@ use types::{
...
-#[derive(PartialEq)]
 struct LocalValidator {
     validator_dir: ValidatorDir,
     voting_keypair: Keypair,
 }
 
+/// We derive our own `PartialEq` to avoid doing equality checks between secret keys.
+///
+/// It's nice to avoid secret key comparisons from a security perspective, but it's also a little
+/// risk when it comes to `HashMap` integrity (that's why we need `PartialEq`).
```
Comment (kirk-baird) [inline validator_client/src/validator_store.rs:26]: risky?

Comment (paulhauner) [reply validator_client/src/validator_store.rs:26]: I swear I fixed this, thanks!

File: crypto/bls/src/secret_key.rs (hunk 1 above)
Comment (paulhauner) [reply crypto/bls/src/secret_key.rs:61]: Yep good call, deleted it in an (unsuccessful) attempt to fully remove `.as_bytes()`, forgot to add it back.

**Pull request merged**